source/module_hsolver/diago_david.cpp

```diff
@@ -53,11 +55,19 @@ void DiagoDavid<T, Device>::diag_mock(hamilt::Hamilt<T, Device>* phm_in,
                                            Real* eigenvalue_in)
 {
     if (test_david == 1)
+    {
         ModuleBase::TITLE("DiagoDavid", "diag_mock");
+    }
     ModuleBase::timer::tick("DiagoDavid", "diag_mock");
 
-    assert(DiagoDavid::PW_DIAG_NDIM > 1);
-    assert(DiagoDavid::PW_DIAG_NDIM * psi.get_nbands() < psi.get_current_nbas() * GlobalV::NPROC_IN_POOL);
+    assert(this->david_ndim > 1);
+
+#ifdef __MPI
+    assert(this->david_ndim * psi.get_nbands() < psi.get_current_nbas() * diag_comm.nproc);
```

Comment:
if MPI is  not defined, the nproc should be 1? then you don't need to write #ifdef __MPI